src/slurm_plugin/instance_manager.py

```diff
@@ -695,12 +697,18 @@ def _scaling_for_jobs_multi_node(
             ),
         )
 
+        # Avoid a job level launch if scaling strategy is BEST_EFFORT
+        # The scale all-in launch has been performed already hence from this point we want to skip the extra
+        # job level launch of instances for jobs that are unable to get the needed capacity from the all-in scaling
+        skip_launch = scaling_strategy == ScalingStrategy.BEST_EFFORT
+
```

Comment:
Unless there are other side effects that I'm missing, the following call seems in charge of making an extra attempt to provision new instances for a job if we weren't able to retrieve them with the first `all-in` launch and IF the strategy IS NOT `best-effort`.

If this is the case, can we simply avoid this call instead of creating a new boolean flag and passing it through the whole call-chain ?

something like 

```
if scaling_strategy != ScalingStrategy.BEST_EFFORT:
  self. _scaling_for_jobs(
  ...
 # else skip call because ...
```

Reply:
There is a side-effect: https://github.com/aws/aws-parallelcluster-node/pull/568#discussion_r1348593364